test/issue-1872/test.k

```diff
@@ -0,0 +1,12 @@
+module TEST
+  imports INT
+
+  configuration
+    <k> $PGM:Pgm </k>
+    <exitcode exit=""> 0 </exitcode>
+
+  syntax Pgm ::= "begin" | "even" | "odd"
+  rule begin => ?_:Int
```

Comment:
Is it certain that the odd rule will be the first to apply here?

Reply:
Thanks for reminding me! I changed the definition to remove the `even` branch. Now there is only one branch, but it still has a side condition.